File: python-wrapper/src/python/model/requests/RequestedPermission.py
from typing import Union, Dict
from enum import Enum

class RequestedPermission(Enum):
	MaxRead = "MAX_READ"
	FullRead = "FULL_READ"
	MaxWrite = "MAX_WRITE"
	FullWrite = "FULL_WRITE"
	Root = "ROOT"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'RequestedPermission':
		if data == "MAX_READ":
			return RequestedPermission.MaxRead
		elif data == "FULL_READ":
			return RequestedPermission.FullRead
		elif data == "MAX_WRITE":
			return RequestedPermission.MaxWrite
		elif data == "FULL_WRITE":
			return RequestedPermission.FullWrite
		elif data == "ROOT":
			return RequestedPermission.Root
		else:
			raise Exception(f"{data} is not a valid value for RequestedPermission enum.")

__all__ = ['RequestedPermission']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/requests/RequestedPermission.kt
package com.icure.cardinal.sdk.model.requests

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class RequestedPermission(
	internal val dtoSerialName: String,
) {
	@SerialName("MAX_READ")
	MaxRead("MAX_READ"),

	@SerialName("FULL_READ")
	FullRead("FULL_READ"),

	@SerialName("MAX_WRITE")
	MaxWrite("MAX_WRITE"),

	@SerialName("FULL_WRITE")
	FullWrite("FULL_WRITE"),

	@SerialName("ROOT")
	Root("ROOT"),
}
